zcash_client_backend/src/data_api/wallet.rs

```diff
@@ -546,12 +546,26 @@ where
                     .memo
                     .as_ref()
                     .map_or_else(MemoBytes::empty, |m| m.clone());
-                builder.add_sapling_output(
-                    external_ovk,
-                    *ua.sapling().expect("TODO: Add Orchard support to builder"),
-                    payment.amount,
-                    memo.clone(),
-                )?;
+
+                if ua.sapling().is_some() {
+                    builder.add_sapling_output(
+                        external_ovk,
+                        *ua.sapling().unwrap(),
+                        payment.amount,
+                        memo.clone(),
+                    )?;
+                } else if ua.transparent().is_some() {
+                    if payment.memo.is_some() {
```

Comment:
I wonder if we should even check the transparent receiver if a memo is provided. This error message could be misleading if someone provided an Orchard+Transparent uaddr; I feel like the error needs to somehow communicate that this error is occurring because we're falling back to the transparent receiver. 

We should at least document this behavior. It also points to a potential issue in the UI; the front end needs to check up front whether it can send a memo to a given uaddr before it decides whether to expose the ability to enter a memo. I think that this issue is what Zooko was referring to with this ticket: https://github.com/zcash/librustzcash/issues/910

Reply:
I think, per the discussion in that thread, we should just document this behavior for now.